tests/Spec.hs

```diff
@@ -1384,6 +1384,291 @@ main = hspec $ do
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 2,Branch "refs/pull/2/head",Sha "ab2")
+                        [PullRequestId 1]
+                        False
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\nSpeculatively rebased as 2b2 behind 1 other PR, waiting for CI …"
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\n[CI job :yellow_circle:](example.com/2b2) started."
+        , AIsReviewer (Username "deckard")
+        , ALeaveComment (PullRequestId 3) "<!-- Hoff: ignore -->\nPull request approved for merge by @deckard, waiting for rebase behind 2 pull requests."
+        , ATryIntegrate "Merge #3: ... Performance\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 3,Branch "refs/pull/3/head",Sha "ab3")
+                        [PullRequestId 1, PullRequestId 2]
+                        False
+        , ALeaveComment (PullRequestId 3) "<!-- Hoff: ignore -->\nSpeculatively rebased as 3b2 behind 2 other PRs, waiting for CI …"
+        , ALeaveComment (PullRequestId 3) "<!-- Hoff: ignore -->\n[CI job :yellow_circle:](example.com/3b2) started."
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\nPush to master detected, rebasing again."
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\nPush to master detected, rebasing again."
+        , ALeaveComment (PullRequestId 3) "<!-- Hoff: ignore -->\nPush to master detected, rebasing again."
+        , ATryIntegrate "Merge #1: Improvements...\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 1, Branch "refs/pull/1/head",Sha "ab1")
+                        []
+                        False
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\nRebased as 1b3, waiting for CI …"
+        , ATryIntegrate "Merge #2: ... Of the ...\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 2,Branch "refs/pull/2/head",Sha "ab2")
+                        [PullRequestId 1]
+                        False
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\nSpeculatively rebased as 2b3 behind 1 other PR, waiting for CI …"
+        , ATryIntegrate "Merge #3: ... Performance\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 3,Branch "refs/pull/3/head",Sha "ab3")
+                        [PullRequestId 1, PullRequestId 2]
+                        False
+        , ALeaveComment (PullRequestId 3) "<!-- Hoff: ignore -->\nSpeculatively rebased as 3b3 behind 2 other PRs, waiting for CI …"
+        ]
+
+    it "ignore a push to master from merging a previous pull request" $ do
```

Comment:
Do you mean `prevent` instead of `ignore`? In which case, can you still add `BuildStatusChanged Succeeded` to PRs 2 and 3? Also I feel like this event/state/result fits better than the `does not restart integration after a push to master from a previous merge in the merge train` test case you have a few tests up.